dll/win32/shlwapi/ordinal.c

```diff
@@ -4121,6 +4121,32 @@ BOOL WINAPI IsOS(DWORD feature)
     return FALSE;
 }
 
+#ifdef __REACTOS__
+/*************************************************************************
+ * @  [SHLWAPI.438]
+ */
+HRESULT WINAPI SHLoadRegUIStringA(HKEY hkey, LPCSTR value, LPSTR buf, DWORD size)
+{
+    WCHAR valueW[MAX_PATH], bufferW[MAX_PATH];
+    DWORD sz = ARRAY_SIZE(bufferW) * sizeof(CHAR);
+    HRESULT hr;
+
+    MultiByteToWideChar(CP_ACP, 0, value, -1, valueW, ARRAY_SIZE(valueW));
+    valueW[ARRAY_SIZE(valueW) - 1] = UNICODE_NULL; /* Avoid buffer overrun */
+
+    if (RegQueryValueExW(hkey, valueW, NULL, NULL, (LPBYTE)bufferW, &sz) != ERROR_SUCCESS)
+        return E_FAIL;
+
+    hr = SHLoadIndirectString(bufferW, bufferW, ARRAY_SIZE(bufferW), NULL);
+    if (FAILED(hr))
+        return hr;
+
+    WideCharToMultiByte(CP_ACP, 0, bufferW, -1, buf, size, NULL, NULL);
+    buf[size - 1] = 0; /* Avoid buffer overrun */
```

Comment:
Does the windows version do this?
I _assume_ that it always ensure the thing is zero terminate, but I doubt it also unconditionally writes at the end of the buffer?

Reply:
> Does the windows version do this?

Surely.

> I doubt it also unconditionally writes at the end of the buffer?
`buffer` has its length of `size`. if `size` is non-zero, `&buf[size - 1]` is in `buf`.

Reply:
Zero check added.